rest/route/project_copy.go

```diff
@@ -0,0 +1,81 @@
+package route
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/evergreen-ci/evergreen/rest/data"
+	"github.com/evergreen-ci/evergreen/rest/model"
+	"github.com/evergreen-ci/gimlet"
+	"github.com/pkg/errors"
+)
+
+////////////////////////////////////////////////////////////////////////
+//
+// POST /rest/v2/projects/{project_id}/copy
+
+type projectCopyHandler struct {
+	oldProjectId string
+	newProjectId string
+	sc           data.Connector
+}
+
+func makeCopyProject(sc data.Connector) gimlet.RouteHandler {
+	return &projectCopyHandler{
+		sc: sc,
+	}
+}
+
+func (p *projectCopyHandler) Factory() gimlet.RouteHandler {
+	return &projectCopyHandler{
+		sc: p.sc,
+	}
+}
+
+func (p *projectCopyHandler) Parse(ctx context.Context, r *http.Request) error {
+	p.oldProjectId = gimlet.GetVars(r)["project_id"]
+	p.newProjectId = r.FormValue("new_project")
+	return nil
+}
+
+func (p *projectCopyHandler) Run(ctx context.Context) gimlet.Responder {
+	projectToCopy, err := p.sc.FindProjectById(p.oldProjectId)
+	if err != nil {
+		return gimlet.MakeJSONErrorResponder(errors.Wrapf(err, "Database error finding project '%s'", p.oldProjectId))
```

Comment:
is this a security violation to let people know that a project exists/doesn't exist

Reply:
oooh is it? Would it be safer to just ask for a different project ID or is that still too transparent?

Reply:
Actually, worth mentioning that the existing UI project creation returns "Project already exists" in this case